src/main/java/com/blog/stream/pagination/PreFetchPageSpliterator.java

```diff
@@ -1,154 +0,0 @@
-package com.blog.stream.pagination;
-
-import org.springframework.data.domain.Page;
-
-import java.util.Spliterator;
-import java.util.function.Consumer;
-
-public class PreFetchPageSpliterator<T> implements Spliterator<T> {
```

Comment:
The idea behind this class was in the case where you didnt want to have call the database/source until you were ready to stream.
With the other spliterator you need to provide a count upfront.

Reply:
Yup. I think I did combine the two spliterator concepts into one that can do both depending on if the prefetched information was available.